File: src/CSharp.Repository/Services/UsersRepository.cs
using CSharp.Forms.Data;
using CSharp.Repository.Interfaces;
using Microsoft.EntityFrameworkCore;
using User = CSharp.Forms.Data.Models.User;

namespace CSharp.Repository.Services
{
    public class UsersRepository : IUsersRepository
    {
        private readonly DataContext _dataContext;

        public UsersRepository(DataContext dataContext)
        {
            _dataContext = dataContext;
        }

        public async Task Add(User user)
        {
            _dataContext.Add(user);
            await _dataContext.SaveChangesAsync();
        }

        public async Task Delete(User user)
        {
            var _user = await _dataContext.Users.FindAsync(user.Id);
            if (_user != null)
            {
                _dataContext.Remove(_user);
                await _dataContext.SaveChangesAsync();
            }
        }

        public async Task<User> Get(string username)
        {
            return await _dataContext.Users.Where(x => x.Name == username).FirstOrDefaultAsync();
        }

        public async Task<IEnumerable<User>> GetAll()
        {
            return await _dataContext.Users.ToListAsync();
        }

        public async Task Update(User user)
        {
            var _user = await _dataContext.Users.FindAsync(user.Id);
            if (_user != null)
            {
                _user.Name = user.Name;
                await _dataContext.SaveChangesAsync();
            }
        }

        public async Task<bool> UserExists(string username)
        {
            return await _dataContext.Users.Where(x => x.Name == username).AnyAsync();
        }

    }
}

File: src/VB.Repository/Services/UsersRepository.vb

Imports Microsoft.EntityFrameworkCore
Imports VB.Forms.Data.VB.Forms.Data
Imports VB.Repository.VB.Repository.Interfaces

Namespace VB.Repository.Services

    Public Class UsersRepository
        Implements IUsersRepository

        Private ReadOnly _dataContext As DataContext

        Public Sub New(dataContext As DataContext)
            _dataContext = dataContext
        End Sub

        Public Async Function GetAll() As Task(Of IEnumerable(Of User)) Implements IUsersRepository.GetAll
            Return Await _dataContext.Users.ToListAsync()
        End Function

        Public Async Function [Get](username As String) As Task(Of User) Implements IUsersRepository.Get
            Return Await _dataContext.Users.Where(Function(x) x.Name = username).FirstOrDefaultAsync()
        End Function

        Public Async Function Update(user As User) As Task Implements IUsersRepository.Update
            Dim _user = Await _dataContext.Users.FindAsync(user.Id)
            If _user IsNot Nothing Then
                _user.Name = user.Name
                Await _dataContext.SaveChangesAsync()
            End If
        End Function

        Public Async Function Delete(user As User) As Task Implements IUsersRepository.Delete
            Dim _user = Await _dataContext.Users.FindAsync(user.Id)
            If _user IsNot Nothing Then
                _dataContext.Remove(_user)
                Await _dataContext.SaveChangesAsync()
            End If
        End Function

        Public Async Function Add(user As User) As Task Implements IUsersRepository.Add
            _dataContext.Add(user)
            Await _dataContext.SaveChangesAsync()
        End Function

        Public Async Function UserExists(username As String) As Task(Of Boolean) Implements IUsersRepository.UserExists
            Return Await _dataContext.Users.Where(Function(x) x.Name = username).AnyAsync
        End Function
    End Class

End Namespace

